### PR TITLE
Added test cases

### PR DESCRIPTION
**Pull request by spencerkclark**
I made a start on adding a test setup here -- feel free to comment on how I've done it.  I tried to keep things minimal.  I basically just added example Runs for AM2 and the idealized moist model and included a basic main script (currently set to compute the mean of the variable `'olr'` in the idealized model).

We still need to determine how to best run multiple tests with one script, and also need to have a way to change the output directory easily for the project (right now it is set to my work directory, so you won't have permission to write to that directory and save calculations done in main.py).

### CODE REVIEW

**Comment by spencerkclark** [comment]
No rush to merge this by the way -- I'm more or less posting this here to generate discussion.

**Comment by spencerahill** [comment]
Thanks for starting on this, which addresses #42.  But this needs to be substantially pruned down.  

I don't think we necessarily need to replicate the whole package structure of `aospy-obj-lib`.  But even if we do, we should have only a few bare bones functions and objects.  I'll comment in-line further.

File: aospy/region.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 """region.py: Region class and region_inst()."""
```
Comment (spencerahill) [inline aospy/region.py:1]: These changes are unrelated to this issue.

File: aospy/test/aospy_test_objs/aospy_user/calcs/numerics.py [hunk 2]
```diff
@@ -0,0 +1,182 @@
+"""Finite differencing and other numerical methods."""
```
Comment (spencerahill) [inline aospy/test/aospy_test_objs/aospy_user/calcs/numerics.py]: This whole module should be omitted.

File: aospy/test/aospy_test_objs/aospy_user/projs/.#am2_2009_yim [hunk 3]
```diff
@@ -0,0 +1 @@
+skc@public1.24097:1435446444
```
Comment (spencerahill) [inline aospy/test/aospy_test_objs/aospy_user/projs/.#am2_2009_yim]: Remove

File: aospy/test/aospy_test_objs/aospy_user/regions/__init__.py [hunk 4]
```diff
@@ -0,0 +1,355 @@
+from aospy.region import Region
```
Comment (spencerahill) [inline aospy/test/aospy_test_objs/aospy_user/regions/__init__.py]: We should have only a few regions.  One global, one that's a simple lat-lon box, one that has multiple lat-lon bounds, one with land mask, and one without land mask.

File: aospy/test/aospy_test_objs/aospy_user/sphere_harm.py [hunk 5]
```diff
@@ -0,0 +1,62 @@
+"""Interface with windspharm/spharm for spherical harmonics analyses."""
```
Comment (spencerahill) [inline aospy/test/aospy_test_objs/aospy_user/sphere_harm.py]: This module should be omitted.

File: aospy/test/aospy_test_objs/aospy_user/variables/__init__.py [hunk 6]
```diff
@@ -0,0 +1,967 @@
+"""Collection of aospy.Var objects for use in my research."""
```
Comment (spencerahill) [inline aospy/test/aospy_test_objs/aospy_user/variables/__init__.py]: There should be only a few variables.  One 2D, one 3D, one at phalf, one at pfull, one that's derived from one or more others, etc.

File: aospy/test/aospy_test_objs/main.py [hunk 7]
```diff
@@ -0,0 +1,213 @@
+#! /usr/bin/env python
```
Comment (spencerahill) [inline aospy/test/aospy_test_objs/main.py]: I don't think we should include all the parsing logic from main.  Can we just create the `Calc` objects and then run them?

**Comment by spencerahill** [comment]
Let me know if any of the comments are unclear.  Ultimately, I think its important (here and elsewhere) to add the minimal amount of code to perform the desired task.  Even if its a temporary fix -- temporary code has a way of sticking around much longer than expected.

Re: the package structure, since we will only have a few of each object type (Model, Run, Var, etc.), I don't think we can just create modules for them at the `aospy_test_objs` level.  Also, let's omit the aospy part and name it `test_objs`.  So e.g. `test_objs/models.py`, `test_objs.runs.py`, etc.

**Comment by spencerahill** [comment]
@spencerkclark , does that all make sense/seem reasonable to you?  Definitely up for further discussion.

> Re: the package structure, since we will only have a few of each object type (Model, Run, Var, etc.), I don't think we can just create modules for them at the aospy_test_objs level. Also, let's omit the aospy part and name it test_objs. So e.g. test_objs/models.py, test_objs.runs.py, etc.

Sorry, this was a typo...the don't should be omitted: I **do** think we can just create modules at this level.

**Comment by spencerkclark** [comment]
Yes, I'm fully on board with pruning things down significantly; sorry for not getting back to you earlier on this (thanks for correcting the typo -- I was a little confused there). 

I'll try and get to this tomorrow if I have time.  Sorry for being quite sloppy in the first iteration (I didn't want to overcommit time to a setup that we wouldn't ultimately use).  Your comments are very helpful.

**Comment by spencerahill** [comment]
Thanks, Spencer.  No rush at all -- just wanted to make sure this didn't fall through the cracks (in that vein, I'll be returning to the AM3 2D bounds issue later tonight).  And I totally hear you re: doing it quick-and-dirty on the first pass.

**Comment by spencerkclark** [comment]
The latest commits cut down a lot on unnecessary code.  Please let me know if there's anything else you'd like me to remove.  I think I might need your help in getting rid of the parsing logic in main.py.  For the simple calculation that is outlined there now, could you give me an example of how to create and execute it without the parsing logic?  Thanks

**Comment by spencerahill** [comment]
Hey I'm really sorry this got forgotten.  What is the status of this nowadays, given some seemingly similar developments you've made in aospy-db?

**Comment by spencerkclark** [comment]
Thanks for the reminder -- yes the progress in aospy-db should make this much easier.  I'll try and get to this next week if I get a chance.

**Comment by spencerahill** [comment]
Closing in favor of #61
